python/ray/util/multiprocessing/pool.py

```diff
@@ -336,6 +442,8 @@ def __init__(self,
         self._initargs = initargs
         self._maxtasksperchild = maxtasksperchild or -1
         self._actor_deletion_ids = []
+        self._registry = []
+        self._registry_hashable = {}
```

Comment:
What is the life time of a Pool? Is it ok to never clear objects in the registry until the Pool is closed?

Reply:
As the functionality in this PR is joblib specific, the Pool will be used as a part of the RayBackend backend for Joblib. Those are usually used as context manager. For example:
```python
    with joblib.parallel_backend('ray'):
        search.fit(X, y)
```
The Pool will be closed on exit from the context. This is the typical usage (one operation), and thus there should be no issues with garbage collection. Some sort of a timer-based eviction could be added, but I do not believe it will be necessary.

Reply:
I see. This makes sense if the usage is limited to the case above. 
Is this feature enabled by being able to import BatchedCalls? Is it possible to unintentionally enable this outside of joblib integration?

Reply:
Only joblib's `BatchedCalls` instances will be impacted, and those are, to my knowledge, used exclusively in joblib Parallel. Any other callables passed to `apply`/`async_apply` will be completely unaffected, with the Pool working as it did before this PR.

Reply:
Sounds good. Maybe add a comment about how _registry and registry_hashable are specific to joblib integration.

Reply:
Correction re closing the Pool - it will be closed by the Parallel object, which means that it will be even stricter than exiting the context (tightly limited to one operation) . Therefore, there should be absolutely no garbage collection issues.